packages/components/src/utils/table/table-data.test.ts

```diff
@@ -0,0 +1,38 @@
+import { beforeEach, describe, expect, test } from 'vitest';
+import { updateRowData } from './table-data';
+import type { TestData } from './test-utils/types';
+
+let data: Array<TestData>;
+let newRowData: TestData;
+
+beforeEach(() => {
+  data = [
+    { name: 'NameData0', value: 'ValueData0' },
+    { name: 'NameData1', value: 'ValueData1' },
+    { name: 'NameData2', value: 'ValueData2' }
+  ];
+  newRowData = { name: 'newDataName', value: 'newDataValue' };
+});
```

Comment:
A followup test should never fail if a previous test fails!
The problem with this let and beforeEach is that let is in this case global for each test, but what if the tests run in parallel and some test do change this data? then an other test fail because of this, which is exactly the reason for flaky and hard to analyze test failures...

Reply:
I see, makes sense. I guess in this case I am adding this data array to the return value of `setupTable` as well. This way, a new array is created for every test. Is this fine with you?